django_project/base/views.py

```diff
@@ -382,16 +403,28 @@ def accept_invite(request, invitation_id):
     except User.DoesNotExist:
         return redirect(f"{reverse('home')}?register_first=true")
 
-    user_profile = UserProfile.objects.get(user=user)
+    user_org = UserOrganisations.objects.filter(
+        user=user,
+        organisation=invitation.organisation
+    ).first()
 
-    if user_profile.organisation == invitation.organisation:
+    if user_org:
         return JsonResponse(
             {
                 "error": "You are already a member of this organization."
             },
             status=400
         )
 
+    # Create a new membership for the user in the organisation
+    UserOrganisations.objects.create(
+        user=user,
+        organisation=invitation.organisation,
+        user_type='member'
+    )
+
+    # Optionally, update the user's profile if necessary
+    user_profile = UserProfile.objects.get(user=user)
     user_profile.organisation = invitation.organisation
```

Comment:
it seems organisation field has been removed from the UserProfile, I think this code will raise an exception.

Reply:
yes it has been removed and replaced with with organisations because a user can belong to more than a single organisation. There are tests inplace to ensure everything is working as expected and no exceptions will be raised